### PR TITLE
postgresql cant have an index with the same name of the table

### PR DESCRIPTION
**Pull request by wizhippo**

### CODE REVIEW

**Comment by gggeek** [comment]
should'nt this go as well in the ini settings? (see how oracle deals with these things)

**Comment by emodric** [comment]
@gggeek Not sure what you mean?

**Comment by gggeek** [comment]
See fe. dbschema.ini in ezflow

**Comment by wizhippo** [comment]
@gggeek I have added one based on your example.  This means no change in the db_schema.dba is needed correct?

**Comment by gggeek** [comment]
Exactly.

I never tested it with postgresql though.

If you have an ezp with postgres, the easy way to check if it works is to go to admin interface => check for updates

**Comment by wizhippo** [comment]
@gggeek Just ran the upgrade check with this change and there are no issues with eztags.  Thanks for the direction.

**Comment by emodric** [comment]
Sorry, but what's the point of adding `dbschema.ini` when the same thing can be achieved directly in SQL file with `CREATE TABLE` statements?

**Comment by gggeek** [comment]
@emodric this is all legacy and will never get properly finished but the overarching idea is:
- only provide dbschema files, complete version
- have users create db sctructures by importing dbschemas
- do not care about how many databases the cms supports, and what are the limitations of each database: only write one dbschema (not entirely true, but can be worked around using dbschema.ini)
- let the cms work out the 'diff' sql when you update the dbschema => easy upgrades
- do not even provide upgrade sqls unless there is the case where data changes

As you can see, these features are so nice that we should have stopped providing sql ages ago

**Comment by andrerom** [comment]
+1

**Comment by emodric** [comment]
@gggeek If that is the case, since we already provide SQL files, and dbschema files are not being widely used, I'd rather just update the SQL files.

**Comment by gggeek** [comment]
Your code, your choice :-)

If you do not plan to keep the dba files up to date, remove them from the repo.

Your tables will break when trying to move data from mysql to postgres using ezsqldumpschema.
And users will just be told to drop your tables when checking the 'update check page'.
But we all know that no-one ever does any of that, right? :-P

**Comment by emodric** [comment]
I didn't say to remove dbschema files. They're useful when checking for db changes in admin.

What I wanted to say is to update all mysql files and dbschema file to have the same names for indexes, to keep them in sync.

**Comment by gggeek** [comment]
Sorry, my bad.
Yes, fixing dba+sql for all dbs and not introducing dbschema.ini is recommended, I thought that you were fixing only the postgres variant in the first place for a specific reason.

**Comment by emodric** [comment]
Yes, this pull request was specific in the first place, but I guess @wizhippo can update it properly to rename all indexes to have `idx_` prefix for example in all files.

**Comment by wizhippo** [comment]
@emodric  Is this what you were thinking?

File: sql/postgresql/schema.sql [hunk 1]
```diff
@@ -11,12 +11,12 @@ CREATE TABLE eztags (
    main_language_id integer not null default 0,
    language_mask integer not null default 0,
    PRIMARY KEY (id),
-   CONSTRAINT remote_id UNIQUE  (remote_id)
+   CONSTRAINT idx_remote_id UNIQUE  (remote_id)
```
Comment (emodric) [inline sql/postgresql/schema.sql]: This should be `idx_eztags_remote_id` too.

**Comment by emodric** [comment]
Yep!

I also made a comment to fix the unique index name in pgsql schema. I know the original was wrong, but since you're at it, you can fix that too :)

**Comment by wizhippo** [comment]
@emodric fixed

**Comment by emodric** [comment]
Thanks, merging!

**Pull request merged**